core/quorum/block_voting.go

```diff
@@ -244,21 +250,46 @@ func (bv *BlockVoting) run(strat BlockMakerStrategy) {
 				case core.TxPreEvent: // tx entered pool, apply to pending state
 					bv.applyTransaction(e.Tx)
 				case Vote:
-					if bv.synced {
-						txHash, err := bv.vote(e.Number, e.Hash)
-						if err == nil && e.TxHash != nil {
-							e.TxHash <- txHash
-						} else if err != nil && e.Err != nil {
-							e.Err <- err
-						} else if err != nil {
-							if glog.V(logger.Debug) {
-								glog.Errorf("Unable to vote: %v", err)
-							}
+					if !bv.canVote() {
+						if e.Err != nil {
+							e.Err <- fmt.Errorf("Node not configured/allowed to vote")
+						}
+						continue
+					}
+
+					if !bv.synced {
+						if e.Err != nil {
+							e.Err <- fmt.Errorf("Node not synced")
+						}
+						continue
+					}
+
+					txHash, err := bv.vote(e.Number, e.Hash)
+					if err == nil && e.TxHash != nil {
+						e.TxHash <- txHash
+					} else if err != nil && e.Err != nil {
+						e.Err <- err
+					} else if err != nil {
+						if glog.V(logger.Debug) {
+							glog.Errorf("Unable to vote: %v", err)
 						}
-					} else {
-						e.Err <- fmt.Errorf("Node not synced")
 					}
+
 				case CreateBlock:
+					if !bv.canCreateBlocks() {
+						if e.Err != nil {
+							e.Err <- fmt.Errorf("Node not configured/allowed to create blocks")
+						}
+						continue
+					}
+
```

Comment:
@bas-vk in a multi-blockmaker configuration, synced will always be false since we aren't passing the singleblockmaker flag so the network will remain on block 0.